examples/flutter_gallery/lib/welcome/home.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+
+import 'step.dart';
+import 'steps/all.dart';
+
+const Color _kWelcomeBlue = Color(0xFF0175c2);
+const int _kAutoProgressSeconds = 10;
+const int _kAutoProgressTransitionMilliseconds = 520;
+const Color footerColor = Color(0xffffffff);
+const TextStyle footerButtonTextStyle = TextStyle(
+  fontSize: 16.0,
+  fontWeight: FontWeight.bold,
+  color: _kWelcomeBlue,
+);
+const double footerButtonHeight = 60.0;
+
+class Welcome extends StatefulWidget {
+  const Welcome({ Key key, this.onDismissed }) : super(key: key);
+  final VoidCallback onDismissed;
+
+  static const String routeName = '/welcome';
+
+  @override
+  _WelcomeState createState() => _WelcomeState();
+}
+
+class _WelcomeState extends State<Welcome> with TickerProviderStateMixin {
+  List<WelcomeStep> _steps;
+  TabController _tabController;
+  PageController _pageController;
+  Timer _autoProgressTimer;
+
+  @override
+  void initState() {
+    super.initState();
+    _steps = <WelcomeStep>[
+      FlutterWelcomeStep(tickerProvider: this),
+      PlaygroundWelcomeStep(tickerProvider: this),
+      DocumentationWelcomeStep(tickerProvider: this),
+      ExploreWelcomeStep(tickerProvider: this),
+    ];
+    _tabController =
+        TabController(initialIndex: 0, length: _steps.length, vsync: this);
+    _pageController = PageController();
+    _autoProgressTimer = _scheduleAutoProgressStepTimer();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final MediaQueryData mediaQueryData = MediaQuery.of(context);
+    final double footerHeight = footerButtonHeight +
+        (Platform.isIOS ? mediaQueryData.padding.bottom : 0.0);
+    return Material(
+      color: _kWelcomeBlue,
+      child: NotificationListener<ScrollNotification>(
+        onNotification: (ScrollNotification notification) {
+          _autoProgressTimer.cancel();
+          _autoProgressTimer = _scheduleAutoProgressStepTimer();
+        },
+        child: Stack(
+          children: <Widget>[
+            Positioned.fill(
+              child: Padding(
+                padding: EdgeInsets.only(bottom: footerHeight),
+                child: PageView.builder(
+                  itemCount: _steps.length,
+                  itemBuilder: (BuildContext context, int index) =>
+                      _steps[index].contentWidget(),
+                  controller: _pageController,
+                  onPageChanged: (int page) {
+                    _tabController.animateTo(page);
+                    _steps[page].animate(restart: true);
+                  },
+                ),
+              ),
+            ),
+            Align(
```

Comment:
Don't forget you can just use SafeArea for stuff like this and in a couple of other areas.